libraries/AP_Proximity/LogStructure.h

```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <AP_Logger/LogStructure.h>
+#include "AP_Proximity.h"
```

Comment:
The problem with doing things this way is that now every header file in ArduPilot will depend on every other header in ArduPilot, as AP_Logger/LogStructure is commonly used.  I can also see us getting into include-loop-hell (you've probably noticed we skirt that here already!

I like the idea of excluding the log structures like this (can't argue with those results!), but we could instead consider moving `HAL_PROXIMITY_ENABLED` out of `AP_Proximity.h` and into `AP_HAL/AP_HAL_Boards.h`.  Making this move came up for other reasons on a DevCall the other week....

Reply:
Moved to `...config.h` files